client-mr/src/main/java/org/apache/hadoop/mapred/SortWriteBufferManager.java

```diff
@@ -321,7 +321,7 @@ ShuffleBlockInfo createShuffleBlock(SortWriteBuffer wb) {
   }
 
   protected void sendCommit() {
-    ExecutorService executor = Executors.newSingleThreadExecutor();
+    ExecutorService executor = Executors.newSingleThreadExecutor(ThreadUtils.getThreadFactory("bufferManagerSend-%d"));
```

Comment:
This is a temporary thread pool. I think it's unnecessary to be a daemon thread pool.WDYT?

Reply:
Yes, I think I made a mistake here, but I think a better way would have been to generate this executor at this object initialization. @jerqi The same as below.

Reply:
MEMORY_LOCALFILE and MEMROY_HDFS don't call commit operation, so we use a temporary thread pool.

Reply:
Got it.